geospaas/base_viewer/views.py

```diff
@@ -56,55 +43,42 @@ def post(self, request, *args, **kwargs):
             self.forms[counter] = element_form_class(request.POST)
 
         self.validation_and_error_check()
-        # modify attributes based on self.form
-        return self.sorting_func(request)
+
+        # modify attributes based on the forms cleaned data
+        self.filtering_func(request)
+        return self.final_rendering(request)
 
     def validation_and_error_check(self):
         for element_forms in self.forms:  # for loop for making clean data by is_valid() method
             element_forms.is_valid()
-            for errorField in element_forms.errors: # temporary error message
+            for errorField in element_forms.errors:  # temporary error message
                 print(f"errorField")
 
-    def sorting_func(self, request):
-        self.ds = CatalogDataset.objects.all()
-        self.filtering_func()
-        return self.final_rendering(request)
-
+    def filtering_func(self, request):
+        ds = CatalogDataset.objects.all()
+        if (request.method == 'POST'):
+            for counter in range(len(self.form_class)):
+                # using the filter function of each form sequentially
+                ds = self.forms[counter].filter(ds)
+            self.ds = ds
+        elif (request.method == 'GET'):
+            self.ds = ds
 
     def final_rendering(self, request):
         ''' Render page based on several forms of data as well as some other context '''
+        self.context = {}
```

Comment:
self.context is not needed. It is better to receive it from self.set_context()
```suggestion
        context = self.set_context()
        eturn render(request, self.main_template, context)
```

Reply:
I have written the `set_context` function and `set_dataset_context` without return statement for easily overriding in the future. This is the reason for this style of coding (to the best of my knowledge). Does `writing the functions of the class with RETURN statement` allow us to override it in the future for ADASviewer without using append in the child class? I don't think so. 

Therefore, it is better to write them in this way (consequently save the results in the self.context[....]) and in the future the child class will be written like this:
```python
def set_dataset_context(self):
        super().set_dataset_context()
        setself.context['additional_context'] = self.blablabla
```